### PR TITLE
refactor!: Replace `EigenStepper` extension list with single extension

### PR DESCRIPTION
**Pull request by andiwand**
Since we always only use one extension in the stepper and there are cases where it is beneficial to only worry about the state of a single extension I propose to remove the `Auctioneer` and the `StepperExtensionList`.

This came up during working on the implementation of the covariance transport where our two default extensions would have to interact via state in the `EigenStepper` which is rather a mess. My proposed solution would move the state consistently to the extension.

**issues**
- https://github.com/acts-project/acts/issues/2868

### CODE REVIEW

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [160b57cb67c62b80381693f8ddf30899e88c1081]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43047503) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8047885)!

File: Core/include/Acts/Propagator/EigenStepperExtensionBase.hpp
[the remark this answers is not in the document]
Comment (andiwand) [reply Core/include/Acts/Propagator/EigenStepperExtensionBase.hpp:1]: That's a very good suggestion, thanks! I will push this in a moment

**Review by benjaminhuth — DISMISSED** [review]
more than 1k lines net removed. nice :+1:

**Review by benjaminhuth — APPROVED** [review]
Let's get it in!

**Pull request merged**